src/api/orders/orderApiRoutes.js

```diff
@@ -23,15 +25,33 @@ app.use(function* getUser(next) {
 
 app.use(function* setQueries(next) {
     this.orderQueries = orderFactory(this.client);
+    this.productQueries = productFactory(this.client);
 
     yield next;
 });
 
 app.use(koaRoute.post('/', function* postUserOrder(next) {
-    this.data = yield coBody(this);
-    this.data.reference = uuid.v1();
-    this.data.customer_id = this.userData.id;
-    this.data.date = new Date();
+    const orderData = yield coBody(this);
+    const productQueries = this.productQueries;
+    const products = yield orderData.products.map(co.wrap(function* getProduct(p) {
```

Comment:
I haven't been able to use generators directly as the callback for `map` functions or promises.